### PR TITLE
Acceptance tests for upload lambda

### PR DESCRIPTION
**Pull request by gcampo88**
Followed up on `feat/upload-signing-tests` and added a currently-failing test for CORS support (unless there's a clearer way to test that than directly calling OPTIONS on the api?)

Depends on: https://github.com/PRX/Infrastructure/pull/30

### CODE REVIEW

File: test/upload/signature_test.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+   it 'return plain text media type' do
+     connection = Excon.new(CONFIG.UPLOAD_HOST)
+     resp = connection.get(query: {to_sign: 'hello'})
+     resp.headers['content-type'].must_equal 'text/plain'
+   end
+
+   it 'return to correct signature' do
+     connection = Excon.new(CONFIG.UPLOAD_HOST)
+     resp = connection.get(query: {to_sign: 'test'})
```
Comment (farski) [inline test/upload/signature_test.rb]: @gcampo88 This should be hitting /sign, right? (which can't be in UPLOAD_HOST, as we are now only using those for FQD's)

File: test/upload/signature_test.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+require 'test_helper'
+
+ describe 'upload-signature' do
+   it 'returns 400 with nothing to sign' do
+     resp = Excon.get(CONFIG.UPLOAD_HOST)
```
Comment (farski) [inline test/upload/signature_test.rb]: Might be good to explicitly test this with an empty `to_sign=` param, or possibly both

Comment (gcampo88) [inline test/upload/signature_test.rb]: 👍

File: test/upload/signature_test.rb (hunk 1 above)
Comment (gcampo88) [inline test/upload/signature_test.rb]: yup, /signature. amended!

**Comment by gcampo88** [comment]
back to you @farski !

File: test/upload/signature_test.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+   end
+
+   it 'return to correct signature' do
+     resp = connection.get(query: {to_sign: 'test'})
+     resp.body.to_s.must_equal "TyhhPs0RA37JFn+0oWNdm25HgBc="
+   end
+
+   it 'enables CORS' do
+     resp = Excon.options(CONFIG.UPLOAD_HOST)
+     resp.status.wont_equal 403
```
Comment (farski) [inline test/upload/signature_test.rb]: @gcampo88  This doesn't seem to really test CORS, just that an `OPTIONS` request isn't `403`. Even a `500` or `404` would pass. Perhaps it should be a bit more explicit in testing the headers that get returned?

Comment (gcampo88) [inline test/upload/signature_test.rb]: 👍

**Comment by pkarman** [comment]
is this PR still relevant?

**Comment by pkarman** [comment]
@gcampo88 @farski can you take a look at this PR and indicate whether it should stay open `In Progress` or get closed?

**Comment by farski** [comment]
I guess In Progress? It's can't be merged as-is afaict and these changes should exist.

**Comment by gcampo88** [comment]
Per meeting on 9/25, we're adding this to Teresa and opening tickets for the relevant work:
https://github.com/PRX/Infrastructure/issues/118
https://github.com/PRX/meta.prx.org/issues/48

**Comment by gcampo88** [comment]
@farski tests pass now with the CF-generated upload host in the .env! 🎉

**Comment by farski** [comment]
@gcampo88 should I merge this, or should we wait until we get custom domain/etc done?

**Comment by gcampo88** [comment]
@farski your call! We'll need to add an UPLOAD_HOST secret to meta, could wait til custom domain to avoid adding then changing. But that's a tiny cost.

**Comment by gcampo88** [comment]
ok, per discussion, let's merge this tomorrow! (though GH still thinks it's blocked)

**Pull request merged**